README.md

```diff
@@ -111,6 +111,66 @@ When running on cluster node, you will also need to distribute this keytab, belo
 
 When Spark application is started, it will transparently track the execution plan of submitted SQL/DF transformations, parse the plan and create related entities in Atlas.
 
+Spark models vs Hive models
+====
+
+SAC classifies table related entities with two different kind of models: Spark / Hive.
+
+We decided to determine which models SAC should use via taking realistic condition: whether Hive Hook can take care of table related entities or not. For creating Hive entities, SAC is not the best one to fill out information on entities - so we let Hive to do its own.
+
+As SAC cannot check whether the condition is properly made, SAC relies on precondition: if Spark session is connected to the "remote HMS" (via thrift), SAC assumes Hive Hook is already set up in remote HMS and Hive Hook will take care of entities creation.
+
+In other words, SAC assumes table entities are being created in Hive side and just refers these entities via object id:
+
+* The value of "spark.sql.catalogImplementation" is set to "hive"
+* The value of "hive.metastore.uris" is set to non-empty
+
+For other cases, SAC will create necessary table related entities as Spark models (even it uses Hive metastore).
+
+One exceptional case is HWC - for HWC source and/or sink, SAC will not create table related entities and always refer to Hive table entities via object id.
+
+Known Limitations (Design decision)
+====
+
+> SAC only supports SQL/DataFrame API (in other words, SAC doesn't support RDD).
+
+SAC relies on query listener to retrieve query and examine the impacts. We may not investigate on supporting RDD for future unless strongly necessary.
+
+> All "inputs" and "outputs" in multiple queries are accumulated into single "spark_process" entity when there're multple queries running in single Spark session.
+
+Unfortunately, we got lost on history for this. We assume this approach could concentrate overall impacts on the single Spark process for who ran for when, but it's not pretty clear why application id was taken.
```

Comment:
I am not sure if its ideal to explicitly state `we got lost on history for this` in the README. Maybe just state the current behavior and mention the alternative approach that could be considered in future.

Reply:
Maybe also state the effect of the current approach. (same batch or streaming queries run in a different spark application will create separate lineage)?

Reply:
I agree that's inappropriate statement. Will remove.

Btw, the case end users mostly concerning is "inputs" and "outputs" being accumulated and the overall graph may look ugly. Ideally we need to explain why we take this approach, but we are not clear about this either, so OK to just state it.

> same batch or streaming queries run in a different spark application will create separate lineage

Agreed for streaming queries but don't get it for batch. Can batch run multiple times with same query ID?